lib/data_type.ml

```diff
@@ -65,6 +65,7 @@ module Typed = struct
     | Float64 : float t
     | Utf8 : string t
     | Binary : string t
+    | Date : Common.Naive_date.t t
```

Comment:
This is a fair point; this is mostly historical (when I originally implemented the accessors this is meant to replace, I exposed the Rust type directly to OCaml instead of using Date.t), but part of it is also due to how the range of dates that `Date.t` supports is different from that of `Common.Naive_date.t`,[1] so I wanted a way for users to be able to work around that, if it ever mattered.

[1]: For example, `Date.t` only seems to support dates in the range of `0` to `9999`, but I think no such limitation exists for `Naive_date.t`.